### PR TITLE
Add doc for equals method in blob

### LINKED ISSUE #310: Document Blob.equals
https://googleapis.dev/java/google-cloud-storage/latest/com/google/cloud/storage/Blob.html

equals method should explain when two blobs are and are not equal

### PR DESCRIPTION
**Pull request by michiboo**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #310 ☕️

### CODE REVIEW

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java [hunk 1]
```diff
@@ -900,6 +900,7 @@ public Builder toBuilder() {
     return new Builder(this);
   }
 
+  /** Returns true if obj's class, toPb and options are equals */
```
Comment (frankyn) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:903]: Please change to:

```
Returns true if obj instance `<pre>{@code Blob.toPb()}</pre>` metadata representation and `<pre>{@code Blob.options}</pre>` instance of StorageOptions are both equal.
```

Also generate javadocs and take a screenshot of the resulting change.

Comment (michiboo) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:903]: image

Hi please see the screen shot above

Comment (frankyn) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: Please inline `<pre>{@code Blob.toPb()}</pre>` and `<pre>{@code Blob.options}</pre>`.

Comment (michiboo) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: @frankyn If I put it inline it won't pass the lint test.... mvn com.coveo:fmt-maven-plugin:format auto format like this.

Comment (frankyn) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: Ah, do this instead:

```
Returns true if obj instance {@code Blob.toPb()} metadata representation and {@code Blob.options} instance of StorageOptions are both equal.
```

Comment (michiboo) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: image

**Comment by michiboo** [comment]
@frankyn  There is error on Kokoro integration test Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 400 Bad Request
"Bucket is requester pays bucket but no user project provided."

**Pull request merged**